src/Surfnet/ServiceProviderDashboard/Application/Metadata/JsonGenerator.php

```diff
@@ -113,65 +114,42 @@ private function generateDataForNewEntity(ManageEntity $entity, $workflowState)
 
     private function generateDataForExistingEntity(
         ManageEntity $entity,
+        EntityDiff $differences,
         string $workflowState,
         string $updatedPart
     ): array {
         $metadata = [
             'entityid' => $entity->getMetaData()->getEntityId(),
         ];
-
         switch ($updatedPart) {
```

Comment:
I think this can be made more readable (courtesy of PHP EA):

The commented parts, would it be more clear if they were in named functions?

```suggestion
        if ($updatedPart === 'ACL') {
            $metadata += $this->generateAclData($entity);
            return $metadata;
        }

        $metadata          += $differences->getDiff();
        if ($entity->getProtocol()->getProtocol() === Constants::TYPE_SAML) {
            $metadata['metadataurl'] = $entity->getMetaData()->getMetadataUrl();
        }
        // Arp is to be sent in its entirety as it does not support the MERGE WRITE feature
        $metadata['arp']   = $this->arpMetadataGenerator->build($entity);
        $metadata['state'] = $workflowState;
        if ($entity->hasComments()) {
            $metadata['revisionnote'] = $entity->getComments();
        }

        // When publishing to production, the coin:exclude_from_push must be present and set to '1'. This prevents the
        // entity from being pushed to EngineBlock. Once the entity is checked a final time, the flag is set to 0
        // by one of the administrators. If the entity was included for push, we make sure it is not overridden.
        if ($entity->isProduction()) {
            $metadata['metaDataFields.coin:exclude_from_push'] = '1';
        }
        if ($entity->isManageEntity() && !$entity->isExcludedFromPush()) {
            $metadata['metaDataFields.coin:exclude_from_push'] = '0';
        }

        return $metadata;
```

Reply:
Thanks for this feedback. This is one of the places where a partial refactor/re delegation of responsibility is in order. For now I try to change as little as possible here.